platform/pulpcore/app/models/content.py

```diff
@@ -43,6 +43,34 @@ def storage_path(self, name):
     sha384 = models.CharField(max_length=96, blank=False, null=False, unique=True, db_index=True)
     sha512 = models.CharField(max_length=128, blank=False, null=False, unique=True, db_index=True)
 
+    # Digest fields ordered by algorithm strength.
+    DIGEST_FIELDS = (
+        'sha512',
+        'sha384',
+        'sha256',
+        'sha224',
+        'sha1',
+        'md5',
+    )
+
+    def is_equal(self, other):
```

Comment:
Ah, good point. They document their approach here: https://docs.djangoproject.com/en/1.11/ref/models/instances/#django.db.models.Model.__eq__

And I agree it may be better to leave that alone.